osu.Server.Queues.ScoreStatisticsProcessor/Commands/Queue/DeleteImportedHighScoresCommand.cs

```diff
@@ -0,0 +1,107 @@
+    /// zero as it gets. But it does mean that we can no longer make the assertion that legacy imports are chronologically imported.
+    /// I'm not sure how important this is in the first place, because they are never going to be perfectly chronological next to
+    /// non-imported (lazer-first) scores anyway...
+    /// </remarks>
+    [Command("delete-high-scores", Description = $"Deletes already-imported high scores from the {SoloScore.TABLE_NAME} table.")]
+    public class DeleteImportedHighScoresCommand : BaseCommand
+    {
+        /// <summary>
+        /// The high score ID to start deleting imported high scores from.
+        /// </summary>
+        [Argument(0)]
+        public ulong StartId { get; set; }
+
+        private ElasticQueueProcessor? elasticQueueProcessor;
+
+        public async Task<int> OnExecuteAsync(CancellationToken cancellationToken)
+        {
+            ulong lastId = StartId;
+            int deleted = 0;
+
+            Console.WriteLine();
+            Console.WriteLine($"Deleting from {SoloScore.TABLE_NAME} starting from {lastId}");
+
+            elasticQueueProcessor = new ElasticQueueProcessor();
+            Console.WriteLine($"Indexing to elasticsearch queue {elasticQueueProcessor.QueueName}");
+
+            Thread.Sleep(5000);
+
+            using (var conn = Queue.GetDatabaseConnection())
+            {
+                while (!cancellationToken.IsCancellationRequested)
+                {
+                    List<ElasticQueueProcessor.ElasticScoreItem> elasticItems = new List<ElasticQueueProcessor.ElasticScoreItem>();
+
+                    using (var transaction = await conn.BeginTransactionAsync(cancellationToken))
+                    {
+                        var highScores = await conn.QueryAsync<SoloScore>("SELECT * FROM solo_scores WHERE id >= @lastId ORDER BY id LIMIT 500", new { lastId = lastId }, transaction);
+
+                        if (!highScores.Any())
+                        {
+                            Console.WriteLine("All done!");
+                            break;
+                        }
+
+                        elasticItems.Clear();
+
+                        foreach (var score in highScores)
+                        {
+                            if (score.ScoreInfo.LegacyScoreId == null)
+                                continue;
+
+                            Console.WriteLine($"Deleting {score.id}...");
+                            await conn.ExecuteAsync("DELETE FROM solo_scores_performance WHERE score_id = @id; DELETE FROM solo_scores WHERE id = @id", score, transaction);
+                            await conn.ExecuteAsync("DELETE FROM solo_scores_legacy_id_map WHERE ruleset_id = @ruleset_id AND old_score_id = @legacy_score_id", new
+                            {
+                                ruleset_id = score.ruleset_id,
+                                legacy_score_id = score.ScoreInfo.LegacyScoreId
+                            }, transaction);
+
```

Comment:
Legacy scores don't actually get a row in here (since they aren't processed):

![iTerm 2023-08-30 at 10 56 17](https://github.com/ppy/osu-queue-score-statistics/assets/191335/4852953a-6deb-47d8-9083-ec29dd35c815)

https://github.com/ppy/osu-queue-score-statistics/blob/168932fe529a800ca4e13f7c047cd448f6b57d14/osu.Server.Queues.ScoreStatisticsProcessor/ScoreStatisticsProcessor.cs#L55-L65